src/collections/maple_tree/mod.rs

```diff
@@ -200,18 +213,90 @@ where
                 M::Some(v) => {
                     return Some(v);
                 }
-                M::None => {}
+                M::None => panic!(
+                    "SparseLeaf - remove() None value found associated to a valid key, aborting."
+                ),
             }
         }
+
         None
     }
 
+    // This function is used to help verify the validity of the entire tree
+    // this function returns true if all keys within the SparseLeaf are within the bounds
+    // min to max or equal to min or max or the SparseLeaf node is empty
+    // otherwise this returns false
+    pub fn check_min_max(&mut self, min: &K, max: &K) -> bool {
+        for i in 0..CAPACITY {
+            match &self.key[i] {
+                M::Some(v) => {
+                    if v >= min && v <= max {
+                        continue;
+                    } else {
+                        return false;
+                    }
+                }
+                M::None => {
+                    continue;
+                }
+            }
+        }
+
+        return true;
+    }
+
     // We need to sort *just before* we split if required.
+    // This function implements selection sort for a SparseLeaf
+    // creates a new SparseLeaf struct, inserts the minimal values
+    // one by one then overwrites the old struct with the new one
+    //
+    // we create the new node so we don't have to deal with the None values
+    // being in-between values otherwise the code would be more complex to handle
+    // compacting the values and then sorting or vice versa so there is no gaps
+    // between actual keys in the underlying array
+    pub fn sort(&mut self) {
+        let mut smallest_key_index: usize;
+        let mut sl: SparseLeaf<K, V> = SparseLeaf::new();
+        let mut sl_index: usize = 0;
+
+        // run once for every key in the sparseLeaf
+        for current_index in 0..CAPACITY {
+            smallest_key_index = 0;
+
+            // run a pass over the remaining items to be sorted to find the
+            // entry with the smallest key and swap it for the item at currentIndex
+            for i in 0..8 {
+                match self.key[i] {
+                    M::Some(_) => {
+                        if self.key[i] < self.key[smallest_key_index] {
+                            smallest_key_index = i;
+                        }
+                    }
+                    M::None => continue,
+                }
+            }
+
+            // swap the found element into the new SparseLeaf with the M::None
+            // that is already in the SparseLeaf instead of just using the insert method
+            // on the new SparseLeaf so the sorting function will keep working
+            //
+            // we could also just just insert the values into the new node and set the value of
```

Comment:
This works for now, we can worry about optimisation of this to avoid swaps later :)

Reply:
Okay sounds good.